tests/unit/test_hnsw_tiered.cpp

```diff
@@ -436,3 +436,184 @@ TYPED_TEST(HNSWTieredIndexTest, insertJobAsyncMulti) {
+        ASSERT_EQ(tiered_index->idToSwapJob.size(), 3);
+        jobQ.pop();
+
+        if (is_multi) {
+            // Insert another vector under the label (2) that has not been deleted.
+            GenerateAndAddVector<TEST_DATA_T>(tiered_index->index, dim, 2, 4);
+            // Expect to see both ids stored under this label being deleted (2 and 4), and have both
+            // ids need repair (as the connection between the two vectors is mutual). Also, 2 has an
+            // incoming edge from his other (deleted) neighbor (3)
+            ASSERT_EQ(tiered_index->deleteLabelFromHNSW(2), 2);
+            ASSERT_EQ(jobQ.size(), 3);
+            ASSERT_EQ(((HNSWRepairJob *)(jobQ.front().job))->node_id, 3);
+            ASSERT_EQ(((HNSWRepairJob *)(jobQ.front().job))->level, 0);
+            jobQ.pop();
+            ASSERT_EQ(((HNSWRepairJob *)(jobQ.front().job))->node_id, 4);
+            ASSERT_EQ(((HNSWRepairJob *)(jobQ.front().job))->level, 0);
+            jobQ.pop();
+            // Repair for node id 4.
+            ASSERT_EQ(((HNSWRepairJob *)(jobQ.front().job))->node_id, 2);
+            ASSERT_EQ(((HNSWRepairJob *)(jobQ.front().job))->level, 0);
+            jobQ.pop();
+            ASSERT_EQ(tiered_index->idToSwapJob.size(), 5);
+            ASSERT_EQ(tiered_index->idToRepairJobs.size(), 3);
+            ASSERT_EQ(tiered_index->idToRepairJobs.at(2).size(), 2);
+            ASSERT_EQ(tiered_index->idToRepairJobs.at(3).size(), 1);
+            ASSERT_EQ(tiered_index->idToRepairJobs.at(4).size(), 1);
+        }
+
+        // Test that repair jobs are created for multiple levels.
+        size_t num_elements_with_multiple_levels = 0;
+        size_t vec_id = tiered_index->index->indexSize();
+        while (num_elements_with_multiple_levels < 2) {
+            GenerateAndAddVector<TEST_DATA_T>(tiered_index->index, dim, vec_id, vec_id);
+            if (tiered_index->getHNSWIndex()->element_levels_[vec_id] > 0) {
+                num_elements_with_multiple_levels++;
+            }
+            vec_id++;
+        }
+
+        // Delete the last inserted vector, which is in level 1.
+        ASSERT_EQ(tiered_index->deleteLabelFromHNSW(--vec_id), 1);
+        ASSERT_EQ(tiered_index->getHNSWIndex()->element_levels_[vec_id], 1);
+        auto *level_one_neighbors =
+            tiered_index->getHNSWIndex()->getNodeNeighborsAtLevel(vec_id, 1);
+        ASSERT_EQ(tiered_index->getHNSWIndex()->getNodeNeighborsCount(level_one_neighbors), 1);
+
+        size_t num_repair_jobs = jobQ.size();
+        // There should be at least two nodes to repair, the neighbors of next_id in levels 0 and 1
+        ASSERT_GE(num_repair_jobs, 2);
+        while (jobQ.size() > 1) {
+            // First we should have jobs for repairing nodes in level 0.
+            ASSERT_EQ(((HNSWRepairJob *)(jobQ.front().job))->level, 0);
+            jobQ.pop();
+        }
+        // The last job should be repairing the single neighbor in level 1.
```

Comment:
1 and higher no?

Reply:
In this scenario specifically we expect that the maximal level of both nodes whose max level is higher than 0 is *exactly* 1